onnx/defs/text/defs.cc

```diff
@@ -34,7 +34,11 @@ ONNX_OPERATOR_SET_SCHEMA(
         }));
 
 static const char* StringSplit_doc =
-    R"DOC(StringSplit splits a string tensor's elements into substrings based on a delimiter attribute and a maxsplit attribute. The first output of this operator is a tensor of strings representing the substrings from splitting each input string on the delimiter substring. A integer tensor is also returned representing the number of substrings generated. This output tensor has one additional rank compared to the input to store these substrings, as examples below will illustrate.)DOC";
+    R"DOC(StringSplit splits a string tensor's elements into substrings based on a delimiter attribute and a maxsplit attribute.
+
+The first output of this operator is a tensor of strings representing the substrings from splitting each input string on the delimiter substring. This tensor has one additional rank compared to the input tensor in order to store the substrings for each input element (where the input tensor is not empty). Note that, in order to ensure the same number of elements are present in the final dimension, this tensor will pad empty strings as illustrated in the examples below. Consecutive delimiters are not grouped together and are deemed to delimit empty strings, except if the delimiter is unspecified or is the empty string (""). In the case where the delimiter is unspecified or the empty string, consecutive whitespace characters are regarded as a single separator and leading or trailing whitespace is removed in the output.
+
+The second output tensor represents the number of substrings generated. The maxsplit attribute can be used to limit the number of splits performed. For elements where fewer splits are possible than specified in maxsplit, it has no effect.)DOC";
```

Comment:
minor nit: may be add the following clarification, wasn't immediately clear to me:
```
The maximum number of substrings returned, when `maxsplit` is specified, is `maxsplit+1`.
For input strings where `maxsplit` splits are possible, the last string returned contains the
trailing prefix after the maxsplit-th split.
```